app/src/main/java/com/example/android/codelabs/paging/ui/SearchRepositoriesActivity.kt

```diff
@@ -81,22 +82,30 @@ class SearchRepositoriesActivity : AppCompatActivity() {
     }
 
     private fun initAdapter() {
-        binding.list.adapter = adapter.withLoadStateHeaderAndFooter(
-            header = ReposLoadStateAdapter { adapter.retry() },
-            footer = ReposLoadStateAdapter { adapter.retry() }
-        )
+        val header = ReposLoadStateAdapter { adapter.retry() }
+        val footer = ReposLoadStateAdapter { adapter.retry() }
+
+        binding.list.adapter = ConcatAdapter(header, adapter, footer)
```

Comment:
why do we need this and we can't use `withLoadStateHeaderAndFooter`? The method should do something similar to what you're doing here.